File: app/src/main/java/com/dof100/morsenotifier/k.java
package com.dof100.morsenotifier;

import javax.microedition.khronos.opengles.GL10;

class k {
   int a;
   int b;
   private d c;

   k(int var1, int var2) {
      this.a = var1;
      this.b = var2;
      d var3;
      switch(this.a) {
      case 1:
         var3 = new d(0.0F, -0.25F, 0.5F, 0.25F);
         break;
      case 2:
         var3 = new d(0.0F, -0.25F, 1.5F, 0.25F);
         break;
      default:
         var3 = null;
      }

      this.c = var3;
   }

   void a(GL10 var1) {
      if (this.c != null) {
         this.c.a(var1);
      }

   }
}

File: app/src/main/java/com/dof100/morsenotifier/k.kt
package com.dof100.morsenotifier

import javax.microedition.khronos.opengles.GL10

internal class k(var a: Int, var b: Int) {
  private val c: d?

  init {
    val var3: d?
    var3 = when (a) {
      1 -> d(0.0f, -0.25f, 0.5f, 0.25f)
      2 -> d(0.0f, -0.25f, 1.5f, 0.25f)
      else -> null
    }
    c = var3
  }

  fun a(var1: GL10) {
    if (c != null) {
      c.a(var1)
    }
  }
}
